plugin/evm/vm.go

```diff
@@ -110,26 +111,29 @@ func maxDuration(x, y time.Duration) time.Duration {
 }
 
 // Codec does serialization and deserialization
-var Codec codec.Codec
+var Codec codec.Manager
 
 func init() {
-	Codec = codec.NewDefault()
+	Codec = codec.NewDefaultManager()
+	c := codec.NewDefault()
 
 	errs := wrappers.Errs{}
 	errs.Add(
-		Codec.RegisterType(&UnsignedImportTx{}),
-		Codec.RegisterType(&UnsignedExportTx{}),
+		c.RegisterType(&UnsignedImportTx{}),
+		c.RegisterType(&UnsignedExportTx{}),
 	)
-	Codec.Skip(3)
+	c.Skip(3)
 	errs.Add(
-		Codec.RegisterType(&secp256k1fx.TransferInput{}),
-		Codec.RegisterType(&secp256k1fx.MintOutput{}),
-		Codec.RegisterType(&secp256k1fx.TransferOutput{}),
-		Codec.RegisterType(&secp256k1fx.MintOperation{}),
-		Codec.RegisterType(&secp256k1fx.Credential{}),
-		Codec.RegisterType(&secp256k1fx.Input{}),
-		Codec.RegisterType(&secp256k1fx.OutputOwners{}),
+		c.RegisterType(&secp256k1fx.TransferInput{}),
+		c.RegisterType(&secp256k1fx.MintOutput{}),
+		c.RegisterType(&secp256k1fx.TransferOutput{}),
+		c.RegisterType(&secp256k1fx.MintOperation{}),
+		c.RegisterType(&secp256k1fx.Credential{}),
+		c.RegisterType(&secp256k1fx.Input{}),
+		c.RegisterType(&secp256k1fx.OutputOwners{}),
 	)
+
+	errs.Add(Codec.RegisterCodec(0, c))
```

Comment:
This can just be included in the previous `Add` chain. Also we should use `codecVersion` here rather than `0`.

Reply:
Updated